spacy/language.py

```diff
@@ -1525,7 +1527,7 @@ def from_config(
         # inside stuff like the spacy train function. If we loaded them here,
         # then we would load them twice at runtime: once when we make from config,
         # and then again when we load from disk.
```

Comment:
```suggestion
        # and then again when we load from disk.
        if vocab in (False, None, 0, 1):
            raise ValueError(Errors.E1010(vocab=vocab))
```

With an error something like:

 E1010 = "Invalid value for vocab: {vocab}. Valid values are a 'Vocab' instance or 'True' to create one."

Reply:
This should happen in `__init__`, though, because that's where it is used